### PR TITLE
hwp angle model metadata

### PR DESCRIPTION
**Pull request by ykyohei**
This PR adds new hwp_angle_model metadata.
hwp_angle_model metadata is for tracking the version of hwp_angle_model, it does not add new functionality to apply_hwp_angle_model. site_pipeline.make_hwp_angle_model.py is a script that will generate hwp_angle_model metadata and we will manage the version manually.

I think there can be a much better implementation of versioning hwp_angle_model or make_hwp_angle_model.py, so I start this PR as a draft. For example, versioning a yaml file or text file seems sufficient for this? I would appreciate any suggestions.

We have to merge this PR and update use_this.yaml at the same time.
My plan of merging procedure is to make a hwp_angle_model metadata in my local directory and ask admin to copy metadata files to the shared directory in site-computing.
Then, we wait until hwp_angle_model metadata is synced to NERSC etc.
Finally, we merge this PR and update the use_this.yaml at site-computing.

### CODE REVIEW

**Comment by ykyohei** [comment]
Addressed suggestions. PR for the scripts for hwp_angle_model is https://github.com/simonsobs/pipeline-metadata-scripts/pull/1
@17-sugiyama I would appreciate if you could doble check.

File: sotodlib/hwp/hwp_angle_model.py [hunk 1]
```diff
@@ -1,12 +1,44 @@
...
+        sign.wrap('pid', -1)
+        sign.wrap('offcenter', 1)
+        model.wrap('mechanical_offset_1', np.deg2rad(-1.66 + 90 - 2.29))
+        model.wrap('mechanical_offset_2', np.deg2rad(-1.66 - 90 - 2.29))
+
+    else:
+        raise ValueError('Not supported yet')
+
+    model.wrap('sign_matrix', sign)
+    model.wrap('sign_dependent_offset', np.deg2rad(-1. * 360 / 1140 * 3 / 2))
```
Comment (mhasself) [inline sotodlib/hwp/hwp_angle_model.py:31]: add `'model_version'`

File: sotodlib/hwp/hwp_angle_model.py [hunk 2]
```diff
@@ -23,27 +55,28 @@ def apply_hwp_angle_model(tod, on_sign_ambiguous='fail'):
...
-        'satp3': {'pid': -1, 'offcenter':  1, },
-    }
+    hwp = tod.get('hwp_solution', None)
+    if hwp is None:
+        raise ValueError('hwp_solution is missing')
+    model = tod.get('hwp_angle_model', None)
+    if model is None:
+        logger.warn('hwp_angle_model metadata is missing. '
+                    'Apply default model. This may be old.')
+        model = default_model(telescope)
```
Comment (mhasself) [inline sotodlib/hwp/hwp_angle_model.py:66]: `assert model.model_version == 'v2'`

**Review by mhasself — CHANGES_REQUESTED** [review]
Great!  Just a couple edits to concord with my suggestions in response to https://github.com/simonsobs/pipeline-metadata-scripts/pull/1

**Comment by ykyohei** [comment]
Done and tested it!

**Review by mhasself — APPROVED** [review]
Great, thanks!

**Pull request merged**